manifests/server/config.pp

```diff
@@ -139,11 +139,7 @@
     require => Exec['puppet_server_config-create_ssl_dir'],
   }
 
-  file { "${::puppet::server::ssl_dir}/private_keys/${::puppet::server::certname}.pem":
-    owner => $::puppet::server::user,
-    group => $::puppet::server::group,
-    mode  => '0640',
-  }
+  ensure_resource('file', "${::puppet::server::ssl_dir}/private_keys/${::puppet::server::certname}.pem", {ensure => file})
```

Comment:
The user/group/mode is important for a default Foreman setup for the foreman-proxy to be able to re-use the key for its TLS communication. From your issue I guess you are defining the content elsewhere?